lib/active_record_shards/connection_switcher.rb

```diff
@@ -153,8 +155,31 @@ def shard_names
       config[SHARD_NAMES_CONFIG_KEY] || []
     end
 
+    if ActiveRecord::VERSION::MAJOR >= 5
+      def connection_specification_name
+        name = current_shard_selection.resolve_connection_name(sharded: is_sharded?, configurations: self.configurations)
+
+        raise "No configuration found for #{name}" unless configurations[name] || name == "primary"
+        name
+      end
+    end
+
     private
 
+    if ActiveRecord::VERSION::MAJOR >= 5
+      def probe_pool
+        # see if we've connected before
```

Comment:
maybe expand this comment to "see if we've connected before.  If not, call `establish_connection` so that activerecord can resolve `connection_specification_name` to an ARS connection